lsp_test.go

```diff
@@ -47,6 +49,361 @@ func TestDocumentOpen(t *testing.T) {
+									Character: 4,
+								},
+								End: lsp.Position{
+									Line:      2,
+									Character: 5,
+								},
+							},
+						},
+						{
+							Message:  "terramate schema error",
+							Severity: lsp.DiagnosticSeverityError,
+							Range: lsp.Range{
+								Start: lsp.Position{
+									Line:      4,
+									Character: 2,
+								},
+								End: lsp.Position{
+									Line:      4,
+									Character: 3,
+								},
+							},
+						},
+						{
+							Message:  "terramate schema error",
+							Severity: lsp.DiagnosticSeverityError,
+							Range: lsp.Range{
+								Start: lsp.Position{
+									Line:      6,
+									Character: 1,
+								},
+								End: lsp.Position{
+									Line:      6,
+									Character: 10,
+								},
+							},
+						},
+						{
+							Message:  "terramate schema error",
+							Severity: lsp.DiagnosticSeverityError,
+							Range: lsp.Range{
+								Start: lsp.Position{
+									Line:      11,
+									Character: 1,
+								},
+								End: lsp.Position{
+									Line:      11,
+									Character: 2,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "multiple errors in the workspace",
+			change: change{
+				file: "terramate.tm",
+				text: "",
+			},
+			layout: []string{
+				`f:bug1.tm:terramate {
+					a = 1
+					b = 2
+				}	
+					`,
+			},
+			want: []WantDiagParams{
+				{
+					URI: "bug1.tm",
+					Diagnostics: []WantDiag{
+						{
+							Message:  "terramate schema error",
+							Severity: lsp.DiagnosticSeverityError,
+							Range: lsp.Range{
+								Start: lsp.Position{
+									Line:      1,
+									Character: 5,
+								},
+								End: lsp.Position{
+									Line:      1,
+									Character: 6,
+								},
+							},
+						},
+						{
+							Message:  "terramate schema error",
+							Severity: lsp.DiagnosticSeverityError,
+							Range: lsp.Range{
+								Start: lsp.Position{
+									Line:      2,
+									Character: 5,
+								},
+								End: lsp.Position{
+									Line:      2,
+									Character: 6,
+								},
+							},
+						},
+					},
+				},
+				{
+					URI:         "terramate.tm",
+					Diagnostics: []WantDiag{},
+				},
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			f := test.Setup(t, tc.layout...)
+			f.Editor.CheckInitialize()
+
+			f.Editor.Change(tc.change.file, tc.change.text)
+			for i := 0; i < len(tc.want); i++ {
+				want := tc.want[i]
+
+				// fix the wanted path as it depends on the sandbox root.
+				want.URI = uri.File(filepath.Join(f.Sandbox.RootDir(), string(want.URI)))
+				select {
+				case gotReq := <-f.Editor.Requests:
+					assert.EqualStrings(t, lsp.MethodTextDocumentPublishDiagnostics,
+						gotReq.Method())
+
+					var gotParams lsp.PublishDiagnosticsParams
+					assert.NoError(t, json.Unmarshal(gotReq.Params(), &gotParams))
+
+					assert.EqualInts(t,
+						len(gotParams.Diagnostics), len(want.Diagnostics),
+						"number of diagnostics mismatch")
```

Comment:
Maybe add the actual diagnostics on the message, should help debugging when it fails.